src/frontend/src/optimizer/plan_node/logical_topn.rs

```diff
@@ -153,9 +156,25 @@ impl ToBatch for LogicalTopN {
     }
 
     fn to_batch_with_order_required(&self, required_order: &Order) -> PlanRef {
-        let new_input = self.input().to_batch();
-        let new_logical = self.clone_with_input(new_input);
-        let ret = BatchTopN::new(new_logical).into();
+        let offset_threshold = 512 * 1024;
+        let ret = if self.offset() < offset_threshold {
```

Comment:
Do we really need this check? I think when we have offset, we should always use topn since we don't need to sort inputs.

Reply:
well, I think the too large topn can not be distributed and parallel. and a large topn might be over the capability of single machine. to make all query processing in the worst case. I think it is needed